datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -669,8 +669,15 @@ impl DisplayAs for HashJoinExec {
                 )
             }
             DisplayFormatType::TreeRender => {
-                // TODO: collect info
-                write!(f, "")
+                let on = self
+                    .on
+                    .iter()
+                    .map(|(c1, c2)| format!("({} = {})", c1, c2))
+                    .collect::<Vec<String>>()
+                    .join(", ");
+
+                writeln!(f, "join_type={:?}", self.join_type)?;
+                writeln!(f, "on=[{}]", on)
```

Comment:
I recommend 
1. we remove `[` and `]`
2. don't print `join_type` if the type is `Inner`